mysql-test/suite/rocksdb/t/use_clock_cache.test

```diff
@@ -0,0 +1,3 @@
+--source include/have_rocksdb.inc
```

Comment:
The test needs some block I/O so that they go through this clock cache.

Also it appears that the test does not fail in a TBB-less build, and what happens here is that NewClockCache returns nullptr which MyRocks happily takes and passes around. This suggests that, if block cache is requested (as it is in "if (!rocksdb_tbl_options->no_block_cache)"), then getting nullptr for it should be a hard error, not silent cache-less operation.

Reply:
No, a TBB less build should fail to compile completely as the definition of TBB specific code in rocksdb offers no compilation alternative for NewClockCache and would result in a basic compilation error. So, This code _requires_ TBB to compile, or not, no in-between.

Reply:
Granted I'm looking at the RocksDB HEAD and not at the specific submodule pointer commit, yet, in clock_cache.h:

#if defined(TBB) && !defined(ROCKSDB_LITE)
#define SUPPORT_CLOCK_CACHE
#endif

In clock_cache.cc:

#ifndef SUPPORT_CLOCK_CACHE

namespace rocksdb {

std::shared_ptr<Cache> NewClockCache(size_t capacity, int num_shard_bits,
                                     bool strict_capacity_limit) {
  // Clock cache not supported.
  return nullptr;
}

}  // namespace rocksdb

#else

... implementation

Reply:
Hmm, yes. I am not sure if I missed this in the diff at the point of the commit or if it was changed in rocksdb after. Either way, it is a mess for us and one that I personally do not wish to expend the time to figure out right now to get this series to pass review. We either take the change as-is, or we skip it entirely for now until some later time when we can properly evaluate and apply, then wait for some web celebrity to complain that we do not embrace the most modern edge case performance bits available. My preference is always stability and consistency across platforms before edge performance.

Reply:
If fixing TBB is too much work, as it seems to be, that's fine, but what I find unacceptable is option being broken silently due to that missing nullptr check. Add that, remove the testcase, no need to patch CMake, file a MYR for TBB, and it's good to go